lib/skate/settings.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Skate.Settings do
+  import Skate.Repo
+
+  alias Skate.Settings.User
+  alias Skate.Settings.UserSettings
+  alias Skate.Settings.VehicleLabel
+
+  @type t :: %__MODULE__{
+          ladder_page_vehicle_label: VehicleLabel.t(),
+          shuttle_page_vehicle_label: VehicleLabel.t()
+        }
+
+  @enforce_keys [
+    :ladder_page_vehicle_label,
+    :shuttle_page_vehicle_label
+  ]
+
+  defstruct [
+    :ladder_page_vehicle_label,
+    :shuttle_page_vehicle_label
+  ]
+
+  @spec get_or_create(String.t()) :: t()
+  def get_or_create(username) do
+    user =
+      insert!(
+        User.changeset(%User{}, %{username: username}),
+        returning: true,
+        conflict_target: [:username],
+        # update a row with no effect so the returning works
+        on_conflict: {:replace, [:username]}
```

Comment:
does it actually do a gratuitous `UPDATE` on conflict or is it smart enough to recognize that that should be a no-op and it does nothing?

Reply:
I don't know. I would guess it does the update.

I might ask the dev channel for help finding a better way to do this.

Reply:
Discussion thread: https://mbta.slack.com/archives/C10EPAU6Q/p1602605004045800

Reply:
The result of that discussion was that `on_conflict` is the right way to do this, and (not what you asked about, but I was wondering) that there isn't a way to insert into both tables at once.